cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -174,8 +194,16 @@ def index_item(item, skip_index=False):
         try:
             with modulestore.branch_setting(ModuleStoreEnum.RevisionOption.published_only):
                 structure = cls._fetch_top_level(modulestore, structure_key)
+                groups_usage_info = None
```

Comment:
This is now adding operations that I believe are specific to course indexing to the base class for both library and course indexing.

Some of it will need to remain, and it is unlikely that there will be content_groups for libraries and so most of this will be skipped. However, I feel that making the logic dependent upon the DOCUMENT_TYPE is a little fragile.

I think I'd prefer to have a base (empty) implementation, say
```
@classmethod
def fetch_group_usage(cls, modulestore, structure):
    return None
```

then the override for CoursewareSearchIndexer looks like:
```
@classmethod
def fetch_group_usage(cls, modulestore, structure):
    groups_usage_info = GroupConfiguration.get_content_groups_usage_info(modulestore, structure).items()
    if groups_usage_info:
        for group in groups_usage_info.values():
            for module in group:
                view, args, kwargs = resolve(module['url'])  # pylint: disable=unused-variable
                module['usage_key_string'] = kwargs['usage_key_string']
    return groups_usage_info
```